### DISCUSSION TITLE
Copy variable under cursor into register

### DISCUSSION THREAD

**Discussion by leo-unglaub** [neovim/neovim#29813] [Q&A] [answered]
Hey,
i would love to hear your ideas and feedback in improving my current workflow. Let's imagine the following PHP code:

```php
public function foobar(): void
{
  $foo = true;
  $bar = 42;
  
  $this->doSomething($foo, $bar, $this->foobar);
}
```
One of the most common things i have to do is to take avariable and add it do a print statement. For example when i have to debug `$foo` i would love to do this:

```php
dump_debug($foo);
```

I dont want to type that all the time, so i have recorded a macro that i have added to my config that takes the word under the cursor, goes to the beginning of the word, goes even one more to the left so that i get the $ as well, then i go up one line and add the `dump_debug($foo);` statement.

This kind of works, but it breaks for `$this->foobar`. So i thought, i have treesitter available, maybe i could use it to not just get the word under the cursor, but also the entire "variable".

Treesitter looks like this:
```treesitter
      arguments: (arguments) ; [30:21 - 47]
       (argument) ; [30:22 - 25]
        (variable_name) ; [30:22 - 25]
         (name) ; [30:23 - 25]
       (argument) ; [30:28 - 31]
        (variable_name) ; [30:28 - 31]
         (name) ; [30:29 - 31]
       (argument) ; [30:34 - 46]
        (member_access_expression) ; [30:34 - 46]
         object: (variable_name) ; [30:34 - 38]
          (name) ; [30:35 - 38]
         name: (name) ; [30:41 - 46]
```

i have the `variable_name` or the `member_access_expression`, but i am at a loss, how can i use that information in my macro?
It would be so nice if you could point me in the right direction.

Thanks and greetings
Leo

**Comment by stCarolas** [comment, marked as answer]
You  could  try to define custom text object with https://github.com/nvim-treesitter/nvim-treesitter-textobjects/tree/master?tab=readme-ov-file#overriding-or-extending-textobjects like `@variable.outer` and use it in macros
